src/main/java/mate/academy/springboot/datajpa/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.springboot.datajpa.service.impl;
+
+import lombok.AllArgsConstructor;
+import mate.academy.springboot.datajpa.repository.CategoryRepository;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+
+    @Override
+    public Category save(Category category) {
+        return categoryRepository.save(category);
+    }
+
+    @Override
+    public Category getById(Long id) {
+        return categoryRepository.getById(id);
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        categoryRepository.deleteById(id);
+    }
+
+    @Override
+    public Category update(Long id, Category category) {
+        return categoryRepository.save(new Category(id, category.getName()));
```

Comment:
we should set id to the category that comes into the method
there is no need to create a new object